packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/RealmSetInternal.kt

```diff
@@ -214,30 +247,73 @@ internal fun <E : BaseRealmObject> ManagedRealmSet<E>.query(
  */
 internal interface SetOperator<E> : CollectionOperator<E, RealmSetPointer> {
 
+    // Modification counter used to detect concurrent writes from the iterator
```

Comment:
Maybe leave a note and reference if this is _"inspired"_ from Java's AbstractList implementation.